timur/lib/client/jsx/components/browser/view_items/plot_item.jsx

```diff
@@ -0,0 +1,41 @@
+// Framework libraries.
```

Comment:
Hm, this component no longer gets the right props, for me? When I open an `rna_seq` page with plots on it, I get several errors in the console and on the server:

![Screenshot from 2020-12-07 14-20-02](https://user-images.githubusercontent.com/4930129/101395017-54de3d00-3897-11eb-9e08-5297ba017e3c.png)

```
ERROR:2020-12-07T19:17:17+00:00 niuslr Caught unspecified error
timur_app_1      | ERROR:2020-12-07T19:17:17+00:00 niuslr PG::InvalidTextRepresentation: ERROR:  invalid input syntax for integer: "null"
timur_app_1      | LINE 1: SELECT * FROM "plots" WHERE "id" = 'null'
timur_app_1      |                                            ^
timur_app_1      | 
```

```
172.23.0.10 - - [07/Dec/2020:19:17:17 +0000] "GET /api/plots/ipi/fetch HTTP/1.1" 200 48299
timur_app_1      | WARN:2020-12-07T19:17:25+00:00 User calling browse#view with params {:project_name=>"ipi", :model_name=>"rna_seq"}
timur_app_1      | WARN:2020-12-07T19:17:25+00:00 User calling manifests#fetch with params {:project_name=>"ipi"}
timur_app_1      | WARN:2020-12-07T19:17:25+00:00 User calling plots#fetch with params {:project_name=>"ipi"}
timur_app_1      | ERROR:2020-12-07T19:17:25+00:00 962qqu Caught unspecified error
timur_app_1      | ERROR:2020-12-07T19:17:25+00:00 962qqu undefined method `to_hash' for nil:NilClass
timur_app_1      | ERROR:2020-12-07T19:17:25+00:00 962qqu /app/lib/server/controllers/browse_controller.rb:6:in `view'
```

Reply:
I'll investigate this further, however, the plots in my default dev instance are mostly non-functional and seem to all be missing a plot configuration (although they have the scripts). I had to copy the plot config from production ipi to test this out.

Reply:
I'm not sure my local plots work either, but I expected the requests to at least be for a plot Id, not `null` or `fetch`...?

Reply:
Sigh, it appears this is the result of bad data. For example, here is the to_hash of the original sample/overview ViewTab record in my timur db:
```{:name=>"overview", :title=>nil, :description=>nil, :panes=>[{:name=>"default", :title=>"Summary", :items=>[{:name=>"patient", :title=>nil, :type=>"magma", :attribute_name=>"patient"}, {:name=>"headshot", :title=>nil, :type=>"magma", :attribute_name=>"headshot"}, {:name=>"tumor_type", :title=>nil, :type=>"magma", :attribute_name=>"tumor_type"}, {:name=>"processed", :title=>nil, :type=>"magma", :attribute_name=>"processed"}, {:name=>"description", :title=>nil, :type=>"magma", :attribute_name=>"description"}, {:name=>"notes", :title=>nil, :type=>"plot", :plot_id=>nil}]}, {:name=>"qc", :title=>"Quality Control", :items=>[{:name=>"qc", :title=>"Quality Control", :type=>"plot", :plot_id=>8}]}]}```

As you can see there is a spurious field "notes" with plot_id "nil" - this is an incorrect piece of data and could never have rendered correctly; this appears to be propagating to the new view. On the other hand you can see the "qc" attribute has a plot_id (with a correctly-configured plot 8), and renders correctly for me.